atest/robot/parsing/suite_names.robot

```diff
@@ -0,0 +1,23 @@
+*** Settings ***
+Documentation     Run testdata and validate that suite names are set correctly
+Suite Setup     Run Tests    ${EMPTY}    misc/suites
+Resource        atest_resource.robot
+# Test Tags       suite_naming
+
+*** Test Cases ***
+Simple Custom Suite Name
+    Should Be Equal As Strings    ${Suite.suites[2].name}    Subsuites
+    Should Be Equal As Strings    ${Suite.suites[2].suites[0].name}    Custom sub1 Name
+
+Simple Default File Suite Name
+    Should Be Equal As Strings    ${Suite.suites[2].name}    Subsuites
+    Should Be Equal As Strings    ${Suite.suites[2].suites[1].name}    Sub2
+
+Cutom Suite Name With Parent init
+    Should Be Equal As Strings    ${Suite.suites[3]}    Parent Suite Name
+    Should Be Equal As Strings    ${Suite.suites[3].suites[0].name}    Sub.Suite.4
+    Should Be Equal As Strings    ${Suite.suites[3].suites[2].name}    SubSuite333
+
+Multi Root Custom Suite Name
+    Should Be Equal As Strings    ${Suite.suites[3].suites[1].name}    SubParentSuite3Name
+    Should Be Equal As Strings    ${Suite.suites[3].suites[1].suites[0].name}    Another Custom Sub Suite
```

Comment:
1. What does "Multi Root" mean here?
2. Add newline at the end of the file.

Reply:
Changed name from `Multi Root' to 'Multi Init' to indicate that the suite name is placed under multiple folders with `__init__.robot` initialization file with setting `Name`. Can it be like that?

Reply:
I don't understand what you mean with "placed under multiple folders". Is the file inside a sub folder that's executed? If yes, then something like `Custom child suite name` would be better. Anyway, the exact name doesn't matter that much.

Reply:
renamed to `Custom Child Suite Name`